src/cogent3/app/data_store_new.py

```diff
@@ -491,6 +491,7 @@ def write(self, *, unique_id: str, data: str) -> DataMember:
         return member
 
     def write_not_completed(self, *, unique_id: str, data: str) -> DataMember:
+        (self.source / _NOT_COMPLETED_TABLE).mkdir(parents=True, exist_ok=True)
         member = self._write(
```

Comment:
@GavinHuttley 
I added this line, because when we run `drop_not_completed`, we delete subdirectory.